packages/core/content-manager/admin/src/hooks/useDocument.ts

```diff
@@ -107,13 +117,31 @@ const useDocument: UseDocument = (args, opts) => {
     ...opts,
     skip: (!args.documentId && args.collectionType !== SINGLE_TYPES) || opts?.skip,
   });
+  const document = data?.data;
+  const meta = data?.meta;
 
   const {
     components,
     schema,
     schemas,
     isLoading: isLoadingSchema,
   } = useContentTypeSchema(args.model);
+  const isSingleType = schema?.kind === 'singleType';
+
+  const getTitle = (mainField: string) => {
+    if (mainField !== 'id' && document?.[mainField]) {
+      // Always use mainField if it's not an id
+      return document[mainField];
+    }
+
+    if (isSingleType && schema?.info.displayName) {
+      // When it's a singleType without a mainField, use the contentType displayName
+      return schema.info.displayName;
+    }
+
+    // Otherwise, use a fallback
+    return 'Untitled';
```

Comment:
Unrelated but does this mean "Untitled" is never localized?

Reply:
Good catch, I'll take a look